doc/src/manual/multi-threading.md

```diff
@@ -388,6 +388,18 @@ threads in Julia:
     This may require some transitional work across the ecosystem before threading
     can be widely adopted with confidence. See the next section for further details.
 
+## [Task Migration](@id man-task-migration)
+
+After a task starts running on a certain thread (e.g. via [`@spawn`](@ref Threads.@spawn) or
+[`@threads`](@ref Threads.@threads)), it may move to a different thread if the task yields.
+
+This means that [`threadid()`](@ref Threads.threadid) should not be treated as constant within a task, and therefore
+should not be used to index into a vector of buffers or stateful objects.
```

Comment:
Shall we say "Instead, prefer making judicious use of `task_local_storage()`, or pass values around as explicit arguments."

Reply:
I think what we should advise is that people use `@spawn` because `@spawn`ed tasks can define and return values. 

That's a lot nicer as an API than having users trying to reason about shared buffers, or them trying to reach into and keep track of `task_local_storage()`s

Reply:
We could keep this PR as is then figure out the recommended patterns in #48542?